Api v5

fivetran-reneeli opened the pull request:
## PR Overview  
**This PR will address the following Issue/Feature:** Update package to align with connector update to API v5

### Submission Checklist  
**Submitter:**  
- [x] Alignment meeting with the reviewer  
- [x] Provide validation details:  
  -  [x] **Validation Steps:** See internal ticket

  - [] **Testing Instructions:** Clear steps for running/testing (e.g., scripts, sample data)  

See internal ticket 

  - [] **Focus Areas:** na

**Reviewer:**  
- [] Confirm submission requirements are met  

### Changelog  
- [] Draft after PR approval

fivetran-reneeli commented:
will hold to regen docs after approval

fivetran-jamie commented on CHANGELOG.md:
```diff
@@ -1,8 +1,22 @@
+[PR #29](https://github.com/fivetran/dbt_apple_search_ads/pull/29) includes the following updates:
+
+## Breaking Changes: Apple Search Ads API v5 schema updates
+- Following the connector updates surrounding the Apple Search Ads API v5 release, the following fields have been added to the upstream `*_report` staging models and will be used to sunset the below fields:
+  - `tap_installs` - replacing `conversions`
+  - `tap_new_downloads` - replacing `new_downloads`
+  - `tap_redownloads` - replacing `redownloads`
```
Again, I know i'm not supposed to look at the changelog rn, but I think it could be good to mention that we are coalescing the new fields with the old to ensure historical data is persisted

fivetran-reneeli replied:
yes updated this! thanks!

fivetran-jamie commented on CHANGELOG.md:
```diff
@@ -1,8 +1,22 @@
+  - `tap_redownloads` - replacing `tap_redownloads`
+  - `total_tap_downloads` - replacing `total_downloads`
+  - `tap_installs` - replacing `conversions`
+- We encourage referencing the new fields, as the fields being replaced will be deprecated by April 2025 in an upcoming release.
```
```suggestion
- **We encourage referencing the new fields, as the fields being replaced will be deprecated by April 2025 in an upcoming release.**
```
or something to make this stick out

fivetran-jamie reviewed (commented):
looks good, largely doc comments!

fivetran-reneeli commented on README.md:
```diff
@@ -56,7 +56,7 @@ Include the following apple_search_ads package version in your `packages.yml` fi
 packages:
   - package: fivetran/apple_search_ads
-    version: [">=0.4.0", "<0.5.0"] # we recommend using ranges to capture non-breaking changes automatically
+    version: [">=0.5.0", "<0.6.0"] # we recommend using ranges to capture non-breaking changes automatically
```
updated and added that note

fivetran-reneeli commented on models/apple_search_ads.yml:
```diff
@@ -43,17 +43,25 @@ models:
         description: Number of taps on ad group on given day.
       - name: new_downloads
-        description: App downloads from new users who have never before installed app of ad group in a given day.
+        description: App downloads from new users who have never before installed app of ad group in a given day. This will be deprecated by April 2025; please refer to the new field which is prefixed with "tap_".
```
updated this and moved new_downloads description to the docs file

fivetran-reneeli commented on models/apple_search_ads.yml:
```diff
@@ -43,17 +43,25 @@ models:
+        description: The sum of new_downloads and redownloads. This will be deprecated by April 2025; please refer to the new field which is prefixed with "tap_".
       - name: conversions
-        description: Total installs resulting from a view or a tap.
+        description: Total installs resulting from a view or a tap. This will be deprecated by April 2025; please refer to the new field which is tap_installs.
```
updated

fivetran-reneeli commented:
Thanks Jamie for the thorough reviews! I moved the field descriptions to the upstream docs file and reference that instead. In addition, I switched "total_tap_downloads" to "tap_total_downloads" to keep with the naming conventions.

I also added the inline comments and made the changelog/readme udpates!

fivetran-jamie reviewed (approved):
Approved with two things:

1. We should use the `tap_total_downloads` doc block in the yml file 
2. a teeny tiny pedantic readme suggestion

fivetran-joemarkiewicz commented on CHANGELOG.md:
```diff
@@ -1,8 +1,23 @@
+  - `tap_installs` - replacing `conversions`
+  - `tap_new_downloads` - replacing `new_downloads`
+  - `tap_redownloads` - replacing `redownloads`
+- Following this, the end models now have the following new fields, which be used to sunset the below fields:
```
This reads a bit strange. Do you mean the following
```suggestion
- Following this, the end models now have the following new fields, which will be used to sunset the below fields:
```

fivetran-joemarkiewicz commented on CHANGELOG.md:
```diff
@@ -1,8 +1,23 @@
+  - `tap_total_downloads` - replacing `total_downloads`
+  - `tap_installs` - replacing `conversions`
+- In order to maintain backwards compatibility and historical data, we have coalesced the new and existing versions of each field.
+- **We encourage referencing the new fields, as the fields being replaced will be deprecated by April 2025 in an upcoming release.**
```
Has this issue been created? Can we create this for the GitHub repo and link it here. Following our new deprecation process.

fivetran-reneeli replied:
https://github.com/fivetran/dbt_apple_search_ads_source/issues/55

fivetran-reneeli replied:
updated!

fivetran-joemarkiewicz commented on README.md:
```diff
@@ -87,7 +87,7 @@ vars:
 
 #### Adding passthrough metrics
-By default, this package will select `taps`, `impressions`, `spend` (aliased from source `local_spend_amount` field), `new_downloads`, `redownloads`, `total_downloads` (sum of `new_downloads` and `redownloads`), and `conversions` (installs resulting from a view or a tap) from the source reporting tables to store into the staging models. If you would like to pass through additional metrics to the  ...
+By default, this package will select `taps`, `impressions`, `spend` (aliased from source `local_spend_amount` field), `new_downloads`, `redownloads`, `total_downloads` (sum of `new_downloads` and `redownloads`), `conversions` (installs resulting from a view or a tap), `tap_new_downloads`, `tap_redownloads`, `tap_installs` (these last 3 fields are replacing `new_downloads`, `redownloads`, `convers ...
```
Was this meant to be included?

fivetran-joemarkiewicz reviewed (changes requested):
Looks good, but a few small comments before approval.
